apps/src/templates/rubrics/AssessmentsAnnouncementDialog.jsx

```diff
@@ -0,0 +1,85 @@
+import React, {useEffect} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {Heading2, BodyTwoText} from '@cdo/apps/componentLibrary/typography';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import AccessibleDialog from '@cdo/apps/sharedComponents/AccessibleDialog';
+import HttpClient from '@cdo/apps/util/HttpClient';
+import {navigateToHref} from '@cdo/apps/utils';
+import i18n from '@cdo/locale';
+
+import announcementImage from './images/ta-assessments-launch-graphic.jpg';
+
+import styles from './rubrics.module.scss';
+
+export default function AssessmentsAnnouncementDialog() {
+  const [dialogOpen, setDialogOpen] = React.useState(true);
+
+  useEffect(() => {
+    if (dialogOpen) {
+      analyticsReporter.sendEvent(EVENTS.TA_RUBRIC_ANNOUNCEMENT_VIEWED);
+    }
+  });
```

Comment:
When you don't pass a dependency array to `useEffect` it runs on every re-render. Is this what you want here? And if it is...why wrap it in a `useEffect` at all?

Reply:
thank you! I was not aware of this subtlety and definitely only want to run it once. good question about why they even allow this syntax.